### PR TITLE
Update device trust docs for v16

### PR DESCRIPTION
**Pull request by codingllama**
Apply a few small changes:

* Document limitations of the `require-trusted-device` role (and update its section)
* Document the need for DEB or RPM Connect packages on Linux

### CODE REVIEW

**Comment by codingllama** [comment]
FYI @rosstimothy

**Review by rosstimothy — APPROVED** [review]

File: docs/pages/access-controls/device-trust/enforcing-device-trust.mdx [hunk 1]
```diff
@@ -41,30 +41,34 @@ can be configured with the `spec.options.device_trust_mode` option and
 applies to the resources in its `allow` rules. It
 works similarly to [`require_session_mfa`](../guides/per-session-mfa.mdx).
 
-<Admonition type="tip" title="v13.3.6+">
-  Teleport version 13.3.6 and above has the preset `require-trusted-device` role.
-  Make sure you update the "allow" rules in the role according to your requirements.
+<Admonition type="warning" title="require-trusted-device role">
+  The preset `require-trusted-device` role does not enforce
+  [Apps](#app-access-support) or [Desktops](#desktop-access-support).
```
Comment (zmb3) [inline docs/pages/access-controls/device-trust/enforcing-device-trust.mdx:46]: Should this say "does not enforce the use of a trusted device for apps for desktops" ?

File: docs/pages/access-controls/device-trust/enforcing-device-trust.mdx [hunk 2]
```diff
@@ -41,30 +41,34 @@ can be configured with the `spec.options.device_trust_mode` option and
...
+  Refer to their corresponding sections for instructions.
 </Admonition>
 
-To enforce authenticated device checks for a specific role, update the role with
-the following:
+The role below is an example of full device trust enforcement. Alternatively,
```
Comment (zmb3) [inline docs/pages/access-controls/device-trust/enforcing-device-trust.mdx:50]: I find it a bit confusing that the previous paragraph says "require-trusted-device does not do full enforcement" followed by an example role named "require-trusted-device" which _does_ do full enforcement.

Can we change the name of the role here?

File: docs/pages/includes/device-trust/troubleshooting.mdx [hunk 3]
```diff
@@ -54,7 +54,8 @@ follow the steps below:
 1. Make sure your device is [registered and enrolled](
    ../../access-controls/device-trust/device-management.mdx#register-a-trusted-device)
 2. Install [Teleport Connect](
-   ../../connect-your-client/teleport-connect.mdx#installation--upgrade)
+   ../../connect-your-client/teleport-connect.mdx#installation--upgrade).
+   Use the DEB or RPM packages on Linux.
```
Comment (zmb3) [inline docs/pages/includes/device-trust/troubleshooting.mdx:58]: I think we should be more explicit here. If the tarball doesn't work, let's say so and explain why.

File: docs/pages/access-controls/device-trust/enforcing-device-trust.mdx (hunk 1 above)
Comment (codingllama) [reply docs/pages/access-controls/device-trust/enforcing-device-trust.mdx:46]: Done.

File: docs/pages/access-controls/device-trust/enforcing-device-trust.mdx (hunk 2 above)
Comment (codingllama) [reply docs/pages/access-controls/device-trust/enforcing-device-trust.mdx:50]: Changed the example to better match require-trusted-device.

File: docs/pages/includes/device-trust/troubleshooting.mdx (hunk 3 above)
Comment (codingllama) [reply docs/pages/includes/device-trust/troubleshooting.mdx]: Done. I think it's a bit awkward, so let me know if you have any suggestions.

**Review by zmb3 — APPROVED** [review]

**Comment by codingllama** [comment]
Thanks for the quick reviews!

**Pull request merged**